pallets/asset-handler/src/lib.rs

```diff
@@ -298,6 +311,51 @@ pub mod pallet {
 			Ok(())
 		}
 
+		/// TODO: Proper Error Handling required, Testing and Benchmarking Pending
+		/// Creates new Asset where AssetId is derived from chain_id and contract Address
+		///
+		/// # Parameters
+		///
+		/// * `origin`: `Asset` owner
+		/// * `network_id`: Network ID of asset being Bridges
+		/// * `identifier_length`: Length of asset identifier length
+		/// * `asset_identifier`: Identifier for a given asset
+		#[pallet::weight(T::WeightInfo::create_asset(1))]
+		pub fn create_thea_asset(
+			origin: OriginFor<T>,
+			network_id: u8,
+			identifier_length: u8,
+			asset_identifier: BoundedVec<u8, ConstU32<1000>>,
+		) -> DispatchResult {
+			T::AssetCreateUpdateOrigin::ensure_origin(origin)?;
+			let mut derived_asset_id = vec![];
+			derived_asset_id.push(network_id);
+			derived_asset_id.push(identifier_length);
+			derived_asset_id.extend(&asset_identifier[0..identifier_length as usize]);
```

Comment:
Please add a check to ensure indexing won't panic

Reply:
Done ✅